src/app/public/modules/back-to-top/back-to-top-adapter.service.ts

```diff
@@ -19,23 +26,67 @@ import {
   takeUntil
 } from 'rxjs/operators';
 
+import {
+  SkyBackToTopComponent
+} from './back-to-top.component';
+
+import {
+  SkyBackToTopType
+} from './models/back-to-top-type';
+
 /**
  * @internal
  */
 @Injectable()
 export class SkyBackToTopDomAdapterService implements OnDestroy {
 
   private ngUnsubscribe = new Subject<void>();
+  private renderer: Renderer2;
 
   constructor(
-    private windowRef: SkyAppWindowRef
-  ) { }
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private applicationRef: ApplicationRef,
+    private injector: Injector,
+    private windowRef: SkyAppWindowRef,
+    private rendererFactory: RendererFactory2
+  ) {
+    // Based on suggestions from https://github.com/angular/angular/issues/17824
+    // for accessing an instance of Renderer2 in a service since Renderer2 can't
+    // be injected into a service.  Passing undefined for both parameters results
+    // in the default renderer which is what we want here.
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
 
   public ngOnDestroy(): void {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
   }
 
+  public createComponentWithinModal(elementRef: ElementRef): ComponentRef<SkyBackToTopComponent> {
```

Comment:
Is this an adapter service a fitting place for this method? When I see a method in an adapter service that has an output type of another component, that's usually a smell that something could be improved. Also, I'm not sure it applies here, but I know when we move to the 5.0 libraries, we changed how the services are provided, and if a service function is returning a component, it *could* end up in a circular dependency. Might have to dig deeper on that to know if its truly a problem in this case.

All that to say, could this method be paired down to only deal with the DOM, and leave the component management to the directive or another service?

Reply:
I mean.... the method is creating a component and then inserting it in the DOM. I understand that we aren't using WebAPI methods here to do that - but we are modifying the DOM here. The only lines in the method that aren't touching DOM stuff are the first two statements and the return. 

I personally would prefer to leave this method here over moving it to the `directive`. I believe historically we have always used the adapter service when working with the Renderer.

Reply:
What about the scenario where we someday might need to access `SkyBackToTopDomAdapterService` from the `SkyBackToTopComponent`? Wouldn't that create a circular dependency?

Reply:
No. Because we aren't doing any of this on initialization. Nothing is telling the adapter to create a back to top component on initialization of anything. So the back to top component could in theory pull in the adapter and tell the adapter to make another back to top but it would be a deliberate thing at that point. The imports would be pointing at each other but nothing functional as they aren't being injected into each other. 

However, this may all change anyway based on Paul's thought below so I'll circle back to this if needed after we figure that out.